### PR TITLE
[#34] Write documentation

### LINKED ISSUE #34: Write documentation
README with usage examples for every command

### PR DESCRIPTION
**Pull request by ChShersh**
Resolves #34

### CODE REVIEW

**Comment by Jigar3** [comment]
@ChShersh When I ran `stack install hit` it created a binary in my `~/.local/bin` named **Hit** and not **hit**. In the docs you mentioned **hit** everywhere and not **Hit**.

**Comment by ChShersh** [comment]
@Jigar3 Thanks for noticing that! Should be fixed now :slightly_smiling_face: Looks like `stack` tried to install this package:

* http://hackage.haskell.org/package/hit

File: README.md [hunk 1]
```diff
@@ -1,17 +1,267 @@
...
-Haskell Git Helper Tool
+**Kowainik Git Workflow Helper Tool.**
 
-## Configure
+You can find the description of the workflow here:
 
-Specify your login:
+* [Kowainik Git Workflow](https://github.com/kowainik/org#workflow)
 
+`hit-on`  provides the `hit` binary with the convenient command-line interface to improve the interaction with [`git`][git] in a compatible way with the described working methods. It saves time for the people who use this workflow on a daily basis and helps beginners to expand the insight of the core VCS processes and ease the cooperation work with the team during development.
```
Comment (kahlil29) [inline README.md]:
```suggestion
`hit-on`  provides the `hit` binary with a convenient command-line interface to improve the interaction with [`git`][git] in a compatible way with the described working methods. It saves time for people who use this workflow on a daily basis, helps beginners expand their insight of the core VCS processes and makes collaboration between team members easier during development.
```

File: README.md [hunk 2]
```diff
@@ -1,17 +1,267 @@
...
+
+### Prerequisites
+
+To start using `hit` make sure that you have the following tools installed on your machine:
+
+* [ `git`][git] — `hit` is a wrapper around `git`
+
+### Installation
+
+There are several methods to install `hit` tool. You can choose the one that you are most comfortable with.
```
Comment (kahlil29) [inline README.md]:
```suggestion
There are several methods to install the `hit` tool. You can choose the one that you are most comfortable with.
```

File: README.md [hunk 3]
```diff
@@ -1,17 +1,267 @@
...
+
+#### Download from releases
+
+You can download the `hit` binary directly from the GitHub releases:
+
+* [`hit` releases](https://github.com/kowainik/hit-on/releases)
+
+After downloading, make it executable and copy it under a convenient location, for example:
```
Comment (kahlil29) [inline README.md]:
```suggestion
After downloading, make it executable and copy it to a convenient location, for example:
```

File: README.md [hunk 4]
```diff
@@ -1,17 +1,267 @@
...
+
+```shell
+chmod +x hit-linux
+mv hit-linux ~/.local/bin/hit
 ```
+
+#### From the sources
```
Comment (kahlil29) [inline README.md]:
```suggestion
#### Build from source
```

File: README.md [hunk 5]
```diff
@@ -1,17 +1,267 @@
...
+    ```shell
+    cd hit-on
+    ```
+
+3. Install the project with one of the build tools
+  * [Cabal](https://www.haskell.org/cabal/users-guide/)
+     ```shell
+        cabal new-install hit-on
+     ```
+    **Note:** make sure you have `~/.cabal/bin` in the $PATH
```
Comment (kahlil29) [inline README.md]:
```suggestion
    **Note:** make sure you have `~/.cabal/bin` in your $PATH
```

File: README.md [hunk 6]
```diff
@@ -1,17 +1,267 @@
...
+  * [Stack](https://docs.haskellstack.org/en/stable/README/)
+     ```shell
+        stack install hit-on
+     ```
+4. Make sure the `hit` is installed:
```
Comment (kahlil29) [inline README.md]:
```suggestion
4. Make sure that `hit` is installed:
```

File: README.md [hunk 7]
```diff
@@ -1,17 +1,267 @@
...
+Follow the steps below to configure `hit` :
+
+1. Specify your GitHub login in the global `.gitconfig`
+```shell
 git config --global user.login <your_login>
 ```
+2. **This step is only required if you want to use `hit` with private repositories**.
+    1. [Create OAuth token on GitHub.](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/)
+    2. Copy the generated token.
+    3. Export token as the environment variable
```
Comment (kahlil29) [inline README.md]:
```suggestion
    3. Export token as an environment variable
```

File: README.md [hunk 8]
```diff
@@ -1,17 +1,267 @@
...
+        ```shell
+        export GITHUB_TOKEN=<paste_generated_token_here>
+        ```
+
+## Usage
+
+The best way to demonstrate the power of the `hit` tool on the day-to-day basis with our workflow is to go through the entire workflow step by step solving an ordinary problem of the typical [`git`][git] user.
```
Comment (kahlil29) [inline README.md]:
```suggestion
The best way to demonstrate the power of the `hit` tool on a day-to-day basis with our workflow is to go through the entire workflow step by step, solving an ordinary problem of the typical [`git`][git] user.
```

File: README.md [hunk 9]
```diff
@@ -1,17 +1,267 @@
...
+
+> Here we assume that you work with `origin` remote with the main branch set to `master`.
+
+### hit hop
+
+When you want to start working on the new issue, make sure you're using the latest version of your project. As a `git` user you may use the following commands:
```
Comment (kahlil29) [inline README.md]:
```suggestion
When you want to start working on a new issue, you usually want to make sure you're using the latest version of your project. As a `git` user you may use the following commands:
```

File: README.md [hunk 10]
```diff
@@ -1,17 +1,267 @@
...
+
+With `hit` you can just:
+
+```shell
+hit hop
+```
+
+### hit issue
+
+Now you need to decide which issue you should work at. You can use the `hit issue` command to see the full list of all open issues. After choosing the number of the issue, let's say 42, call `hit issue 42` to see the details of that issue.
```
Comment (kahlil29) [inline README.md]:
```suggestion
Now you need to decide which issue you want to work on. You can use the `hit issue` command to see the full list of all open issues. After choosing the number of the issue, let's say 42, call `hit issue 42` to see the details of that issue.
```

File: README.md [hunk 11]
```diff
@@ -1,17 +1,267 @@
...
+<user_login>/<issue_number>-<short_issue_description>
+```
+
+With `git` you can create a branch using the following command:
+
+```shell
+git checkout -b my-login/42-short-desc
+```
+
+`hit` allows to accomplish this task easier:
```
Comment (kahlil29) [inline README.md]:
```suggestion
`hit` allows you to accomplish this task in an easier manner:
```

File: README.md [hunk 12]
```diff
@@ -1,17 +1,267 @@
...
+
+```shell
+hit commit "Implement my feature"
+```
+
+Note that you don't need to keep in mind the current issue number. However, if you want to refresh the context about the issue, use the `hit current` command.
+
+### hit push
+
+After committing your changes locally, you need to push them to the remote repository. It's usually a good practice only to push the current branch.
```
Comment (kahlil29) [inline README.md]:
```suggestion
After committing your changes locally, you need to push them to the remote repository. It's usually a good practice to push only the current branch.
```

File: README.md [hunk 13]
```diff
@@ -1,17 +1,267 @@
...
+
+`git` command for this is a little bit verbose:
```
Comment (kahlil29) [inline README.md]:
```suggestion
The `git` command for this is a little bit verbose:
```

File: README.md [hunk 14]
```diff
@@ -1,17 +1,267 @@
...
+
+However, with `hit` you can just:
+
+```shell
+hit sync
+```
+
+### hit fresh
+
+While you were waiting for the second review round, another pull request was merged to the `master` branch. Now you need to apply the new `master` changes to your local branch.
```
Comment (kahlil29) [inline README.md]:
```suggestion
While you were waiting for the second round of reviews, another pull request was merged to the `master` branch. Now you need to apply the new `master` changes to your local branch.
```

File: README.md [hunk 15]
```diff
@@ -1,17 +1,267 @@
...
+
+`hit` helps you with this as well:
+
+```shell
+hit fix
+```
+
+### hit amend
+
+Oops, you've just realised that you have made a typo in your work! Now you want to fix this problem but without creating new unnecessary commit.
```
Comment (kahlil29) [inline README.md]: Maybe we should mention here that the following commands should be run after fixing the typo. I had to read this a couple of times before I got it. Just a suggestion

File: README.md [hunk 16]
```diff
@@ -231,7 +231,7 @@ hit fix
 
 ### hit amend
 
-Oops, you've just realised that you have made a typo in your work! Now you want to fix this problem but without creating new unnecessary commit.
+Oops, you've just realised that you have made a typo in your work! So you fixed the typo. But now you want to update the remote branch without creating new unnecessary commit.
```
Comment (kahlil29) [inline README.md]: without creating **a** new unnecessary commit

**Pull request merged**